src/wallet/wallet.h

```diff
@@ -30,7 +30,8 @@
 
 #include <boost/shared_ptr.hpp>
 
-extern CWallet* pwalletMain;
+typedef CWallet* CWallet_ptr;
```

Comment:
not sure it make sense either, but if it is done, better naming it to `CWalletRef` to be consistent with `CTransactionRef`